sync/sync.go

```diff
@@ -26,14 +28,16 @@ func (e ErrSyncFailed) Error() string {
 type Synchronizer struct {
 	Blockchain   *blockchain.Blockchain
 	StarknetData starknetdata.StarknetData
+	Gateway      *clients.GatewayClient
```

Comment:
synchronizer shouldnt need a GatewayClient reference.

Reply:
Had to add this cos l1Verifier is called in sync

Reply:
see https://github.com/NethermindEth/juno/pull/561#discussion_r1116848109 then